### DISCUSSION TITLE
Questions about kyuubi asynchronous and restful API function.

### DISCUSSION THREAD

**Discussion by zhichaoleo** [apache/kyuubi#1482] [Q&A] [answered]
Hi Team,

Does kyuubi support asynchronous submission of spark sql? I saw the pinned issue KPIP-1 suggesting that kyuubi should support restful API, in which version can we use it at the earliest?

Thanks

**Comment by yaooqinn** [comment, marked as answer]
> support asynchronous submission of spark sql

if you mean executing SQL statement, it's already supported, but you need to ensure your client-side hive protocol also support this
if you mean open session with engine launching asynchronously, it is coming with v1.4.0(it is voting now, please subscribe to our mail list and help vote it).


> restful API

It's incomplete yet, maybe 1.5.0 if we can get enough help

**Reply by zhichaoleo** [reply]
Got it! Thanks for your help!
